pyomo/contrib/mindtpy/util.py

```diff
@@ -79,7 +79,10 @@ def model_is_valid(solve_data, config):
             set_solver_options(mainopt, solve_data,
                                config, solver_type='mip')
             mainopt.solve(solve_data.original_model,
-                          tee=config.mip_solver_tee, **config.mip_solver_args)
+                          tee=config.mip_solver_tee,
+                          load_solutions=config.mip_solver not in {'appsi_cplex', 'appsi_gurobi'},
+                          **config.mip_solver_args)
+            load_solution_appsi(mainopt, config)
```

Comment:
Instead of what I mentioned below, you could do

```
results = main opt.solve(
    solve_data.original_model,
    tee=config.mip_solver_tee,
    load_solutions=False,
    **config.mip_solver_args
)
if pyo.check_optimal_termination(results):
    solve_data.original_model.solutions.load_from(results)
```

Then, you would not need the `load_solution_appsi` function.

Reply:
If you want to load the solution with termination conditions other than optimal, you can do that as well:

```
results = main opt.solve(
    solve_data.original_model,
    tee=config.mip_solver_tee,
    load_solutions=False,
    **config.mip_solver_args
)
if len(results.solution > 0):
    solve_data.original_model.solutions.load_from(results)
```